### PR TITLE
Add Parsing of tensor constants

### PR DESCRIPTION
**Pull request by eellison**
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #75124
* __->__ #75119
* #75108

Add support for parsing Tensor constants like Double(4, 4) ... by initializing random tensors. This makes saving IR and then parsing it lossy, so I have it toggled as default not on, but is useful in cases like repro-ing Fusions with tensor constants post-freezing.

cc @Krovatkin

Differential Revision: [D35373999](https://our.internmc.facebook.com/intern/diff/D35373999)

### CODE REVIEW

File: torch/csrc/jit/ir/irparser.h [hunk 1]
```diff
@@ -13,17 +14,27 @@ struct Graph;
...
+// throw
+TORCH_API void parseIR(
+    const std::string& str,
+    torch::jit::Graph* graph,
+    bool parse_tensor_constants = false);
 
 /** \brief Parse IR from \p STR constructing the corresponding IR in\ GRAPH.
  *
- * \p VMAP is filled with String to Value pairs allowing to index Values in the
+ * O\p VMAP is filled with String to Value pairs allowing to index Values in the
```
Comment (ZolotukhinM) [inline torch/csrc/jit/ir/irparser.h:27]: Typo.

**Review by ZolotukhinM — APPROVED** [review]
This is supernice, thanks! A couple of remarks below:

File: torch/csrc/jit/ir/irparser.cpp [hunk 2]
```diff
@@ -117,17 +128,21 @@ VarWithType IRParser::parseVarWithType(bool allow_optional) {
...
-  } else {
-    return L.expect(TK_NUMBER).text();
-  }
+    continue_parsing = false;
+    if (L.nextIf('.')) {
+      continue_parsing = true;
+      name += '.';
+    }
+  } while (continue_parsing);
+  return name;
```
[the remark this answers is not in the document]
Comment (eellison) [reply torch/csrc/jit/ir/irparser.cpp:145]: Oop, didnt mean to include that, it's in the next PR https://github.com/pytorch/pytorch/pull/75124

**Comment by eellison** [comment]
@eellison has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D35373999).

**Comment by eellison** [comment]
@eellison has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D35373999).

**Pull request closed without merging**